### DISCUSSION TITLE
Best practice to avoid "dangling transaction"

### DISCUSSION THREAD

**Discussion by fluxxu** [weiznich/diesel_async#49] [General]
https://docs.rs/diesel-async/0.2.0/diesel_async/trait.AsyncConnection.html#method.transaction

> WARNING: Cancling the returned future does currently not close an already open transaction. You may end up with a connection containing a dangling transaction.

Is there any suggestion to avoid this problem, as I understand it, even if I don't actively cancel a future, maybe the framework I'm using, or the library function might do it implicitly?

Thank you.

**Comment by weiznich** [comment]
We do explicitly not provide suggestions how to avoid this problem as this is currently inherent on how rust async ecosystem is designed. That's the reason why this is documented this way.
